python/sidecar/http.py

```diff
@@ -192,20 +195,13 @@ async def create_project(project_name: str, project_path: str = None):
 
 
 @project_api.get("/{project_id}")
-async def get_project(project_id: str):
+async def get_project(project_id: str) -> ProjectDetails:
     """
     Returns the project path and a list of files in the project
     """
     user_id = "user1"
-    project_path = projects.get_project_path(user_id, project_id)
-    project_name = projects.get_project_name(user_id, project_id)
-    filepaths = projects.get_project_files(user_id, project_id)
-    return {
-        "project_id": project_id,
-        "project_path": project_path,
-        "project_name": project_name,
-        "filepaths": filepaths,
-    }
+    response = projects.get_project_files(user_id, project_id)
```

Comment:
why do we need to call `get_project_files` to only retrieve the project details? No need to read and create the filesystem structure for this endpoint.

Also, the docs for this method are wrong.